homework/list.js

```diff
@@ -47,33 +40,39 @@
       li.dataset.noteId = i;
       li.classList.add('note-title');
       li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    this._wrapper.appendChild(ul);
+  },
 
-  function fetchList(afterFetch) {
+  fetchList(afterFetch) {
+	  return new Promise((function(resolve,reject){
     var xhr = new XMLHttpRequest();
     xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
     xhr.responseType = 'json';
     xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
       if (this.readyState === 4 && this.status === 200) {
         var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
+        resolve(listData);
```

Comment:
Good. You have successfully "Promisifed" this method with only one nit: you don't need the `afterFetch` callback anymore, so please remove it.